### PR TITLE
Geolocation

### PR DESCRIPTION
**Pull request by G-Ray**
With these changes, we now have a working Geolocate control https://www.mapbox.com/mapbox-gl-js/api#geolocatecontrol.

Still need to be implemented:
The zoom level is already depending on the accuracy: https://github.com/mapbox/mapbox-gl-js/blob/026bb44f07f089f5e2d0e3b56739bd5af708decc/src/ui/control/geolocate_control.js#L189)
- [x] Modal when enabling the geolocation

### CODE REVIEW

**Comment by G-Ray** [comment]
Maybe we can already merge theses changes as this for now as a first iteration ?

File: src/scss/includes/modals/share.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#share_url_data {
+  display: block;
+  width: 100%;
+  margin-bottom: 5px;
+  padding: 10px 12px;
+  border-radius: 3px;
```
Comment (Julien-laville) [inline src/scss/includes/modals/share.scss]: you can use @mixin card_radius()

File: language/message/fr_FR.po [hunk 2]
```diff
@@ -84,945 +83,1001 @@ msgstr "Toilettes partiellement accessibles en fauteuil roulant"
...
+"accéder à votre localisation pour qu’elle puisse vous l’indiquer. Comme "
+"indiqué dans [notre politique de protection des données], Qwant ne stockera "
+"pas vos informations de géolocalisation. Nous ne voulons pas connaître vos "
+"allers et venues."
 
-#: public/build/javascript/bundle.js:2556
-msgid "SHARE"
-msgstr "PARTAGER"
+#: public/build/javascript/bundle.js:15370
+msgid "I surf in complete tranquility."
```
Comment (Julien-laville) [inline language/message/fr_FR.po]: English version doesn't ring well

File: src/scss/includes/map_theme.scss [hunk 3]
```diff
@@ -96,4 +98,36 @@
...
+}
+
+.mapboxgl-ctrl-icon.mapboxgl-ctrl-geolocate {
+  @extend [class^="icon-"];
+  @extend .icon-pin_geoloc;
+  background-image: none !important;
+  color: $primary_text;
+  font-size: 20px;
+
+  @media(max-width: 640px) {
```
Comment (Julien-laville) [inline src/scss/includes/map_theme.scss]: by habit we put all media_queries together.
media queries should contains classes

File: language/message/fr_FR.po [hunk 4]
```diff
@@ -84,945 +84,992 @@ msgstr "Toilettes partiellement accessibles en fauteuil roulant"
...
+#: public/build/javascript/bundle.js:15354
+msgid "Géolocation not available."
+msgstr "Géolocalisation impossible"
 
-#: public/build/javascript/bundle.js:2547
-msgid "We use MASQ to keep your preference data safe"
-msgstr "Nous utilisons MASQ pour conserver vos préférences en sécurité"
+#: public/build/javascript/bundle.js:15354
+msgid "You did not allow Qwant Maps to access your position."
+msgstr "vous n'avez pas autorisé Qwant Maps à accéder à votre position."
```
Comment (amatissart) [inline language/message/fr_FR.po]: Missing capital letter

File: language/message/fr_FR.po [hunk 5]
```diff
@@ -84,945 +84,992 @@ msgstr "Toilettes partiellement accessibles en fauteuil roulant"
...
+"Comme indiqué dans {privacyPolicyLink} notre politique de protection des "
+"données {closeTag}, Qwant ne stockera pas vos informations de "
+"géolocalisation. Nous ne voulons pas connaître vos allers et venues."
 
-#: public/build/javascript/bundle.js:2556
-msgid "SHARE"
-msgstr "PARTAGER"
+#: public/build/javascript/bundle.js:15373
+msgid "Ok, I've got it."
+msgstr "ok, j'ai compris."
```
Comment (amatissart) [inline language/message/fr_FR.po]: idem

File: src/scss/includes/z_index.scss [hunk 6]
```diff
@@ -32,4 +32,16 @@
...
+  z-index: 4;
+}
+
+.top_bar {
+  z-index: 2;
+}
+
+.service_panel_mobile {
+  z-index: 3;
+}
```
Comment (amatissart) [inline src/scss/includes/z_index.scss]: There is a problem on mobile with these z-index values.  
The "favorites" button stays on top of the input field and the cross is hidden:
image

File: language/message/fr_FR.po [hunk 7]
```diff
@@ -84,945 +84,992 @@ msgstr "Toilettes partiellement accessibles en fauteuil roulant"
...
+#: public/build/javascript/bundle.js:15370
+#, python-brace-format
+msgid ""
+"You need to know where you are? Please allow your application to access your "
+"location so that it can tell you. As stated in our {privacyPolicyLink} "
+"privacy policy {closeTag}, Qwant will not store your geolocation "
+"information. We don't want to know your whereabouts."
+msgstr ""
+"Besoin de savoir où vous êtes ? Veuillez svp autoriser votre application à "
+"accéder à votre localisation pour qu’elle puisse vous l’indiquer.\n"
```
Comment (amatissart) [inline language/message/fr_FR.po]: the newline `\n` is not properly handled by the template. The letter "n" is rendered.

File: language/message/fr_FR.po (hunk 5 above)
Comment (Julien-laville) [reply language/message/fr_FR.po]: Ok, I've got it

File: language/message/fr_FR.po (hunk 7 above)
Comment (G-Ray) [reply language/message/fr_FR.po]: Thank you, I removed it!

**Pull request merged**